### PR TITLE
QL: fix non-attached annotations for newtype branches

### PR DESCRIPTION
**Pull request by erik-krogh**
Fixes this FP: https://github.com/github/codeql/security/code-scanning/77304

### CODE REVIEW

File: ql/ql/src/codeql_ql/ast/Ast.qll [hunk 1]
```diff
@@ -983,6 +983,8 @@ class NewTypeBranch extends TNewTypeBranch, Predicate, TypeDeclaration {
 
   override NewTypeBranchType getReturnType() { result.getDeclaration() = this }
 
+  override Annotation getAnAnnotation() { toQL(this).getAFieldOrChild() = toQL(result) }
```
Comment (tausbn) [inline ql/ql/src/codeql_ql/ast/Ast.qll:986]: Should `getAChild` be updated, or do we not count annotations as children of the nodes they are attached to?

**Review by tausbn — APPROVED** [review]
One question, otherwise this looks good to me. 👍

File: ql/ql/src/codeql_ql/ast/Ast.qll (hunk 1 above)
Comment (erik-krogh) [reply ql/ql/src/codeql_ql/ast/Ast.qll:986]: The base definition of `getAChild` (in `AstNode`) includes `getAnAnnotation()`.   

So `getAnAnnotation()` is part of `getAChild` here in `NewTypeBranch` through the super call.  

I think that answers your question.

Comment (tausbn) [reply ql/ql/src/codeql_ql/ast/Ast.qll:986]: Makes sense, thank you!

**Pull request merged**